### ISSUE TITLE
Only 2 out of 3 monitors are working

### ISSUE THREAD

**Issue by theErikss0n** [swaywm/sway#5315]
## Sway Version:

1.4

##  Debug Log:

[sway.log](https://gist.github.com/theErikss0n/c3079796f1be69670e216db30f919c44)
[swaymsg -t get_outputs](https://gist.github.com/theErikss0n/cde81e2a37c4ea45d27ea2f7f23d5f83)

outputs reference:
DP-4: 4k monitor in the middle of picture below
DP-5: FullHD Monitor on the left
eDP-1: Laptop screen

##  Configuration File:

default config

## Steps to reproduce
1. Have Laptop with one FullHD  and one 4k Monitor connected via dock
2. launch sway from tty with `sway`
3. only 2 out of 3 monitors are working.

![IMG_20200508_121347](https://user-images.githubusercontent.com/38251684/81397484-b682d100-9127-11ea-9b3a-1dee4747586e.jpg)
(Left and right  monitors are working, middle monitor is not)

Also:
1. Disconnect thunderbolt cable from laptop
2. Start sway with `sway`
3. connect dock cable.
4. now laptop screen and my 4k monitor are working, but not my FullHD monitor

Note: Last time it worked for me was sway 1.2.  Some update broke it and i just used the old working version. But now i was forced to upgrade.

**Comment by emersion** [comment]
Can you try setting `WLR_DRM_NO_MODIFIERS=1`?

**Comment by theErikss0n** [comment]
unfortionatly the problem persists.

[sway.log](https://gist.github.com/theErikss0n/1a91f66eec9071d4224b514cd11e8d05)

**Comment by emersion** [comment]
Can you obtain DRM debug logs? See https://github.com/swaywm/wlroots/wiki/DRM-Debugging

```
2020-05-08 13:56:50 - [backend/drm/drm.c:695] Modesetting 'DP-4' with '3840x2160@59996 mHz'
2020-05-08 13:56:50 - [backend/drm/drm.c:591] Initializing renderer on connector 'DP-4'
2020-05-08 13:56:50 - [backend/drm/atomic.c:56] DP-4: Atomic commit failed (modeset): Invalid argument
2020-05-08 13:56:50 - [backend/drm/drm.c:610] Page-flip failed with primary FB modifiers enabled, retrying without modifiers
2020-05-08 13:56:50 - [backend/drm/atomic.c:56] DP-4: Atomic commit failed (modeset): Invalid argument
```

**Comment by theErikss0n** [comment]
[sway drm log](https://gist.github.com/theErikss0n/6acb75aa1e97be26c194d598f3f06a6e)
[dmesg log](https://gist.github.com/theErikss0n/4b20f57128e4c097757591df6c48a39d)

**Comment by emersion** [comment]
```
[15392.711324] i915 0000:00:02.0: [drm:skl_allocate_pipe_ddb [i915]] Requested display configuration exceeds system DDB limitations
[15392.711351] i915 0000:00:02.0: [drm:skl_allocate_pipe_ddb [i915]] minimum required 242/199
[15392.711383] [drm:intel_dump_pipe_config [i915]] [CRTC:51:pipe A] enable: yes [failed]
```

I'm afraid your GPU doesn't support 3 monitors with these resolutions. Picking a lower resolution may make it work.

**Comment by theErikss0n** [comment]
Weirdly the setup with 3 Monitors worked with a previous version of sway and wlroots.

If the CPU couldn't support it, it couldn't have worked before

**Comment by emersion** [comment]
`WLR_DRM_NO_MODIFIERS=1` should restore the old Sway behaviour. The kernel also has changed its checks, so maybe they need to be relaxed.

**Comment by theErikss0n** [comment]
So i got lucky and managed to downgrade sway and some deps and got it working again: 

sway version 1.2
[swaymsg -t get_outputs](https://gist.github.com/theErikss0n/c7368fb3878e32a1e2a345f980e9e6c0)
[swaydrm.log](https://gist.github.com/theErikss0n/d66478365d34043954e361d2b73fd2b0)
[dmesg.log](https://gist.github.com/theErikss0n/f533b0552a37029985deae83e68935db)

**Comment by emersion** [comment]
Oh, so same kernel but doesn't work? Interesting.

Can you try sway+wlroots master? See https://github.com/swaywm/sway/wiki/Development-Setup#compiling-as-a-subproject

**Comment by theErikss0n** [comment]
So things get a little weird.

First things first: All 3 monitors work with the current master.
[swaydrm.log](https://gist.github.com/theErikss0n/9ea6655ef5d93d5191cb71c6c664cdb4)
[dmesg.log](https://gist.github.com/theErikss0n/166a17b3b5dc83b2e847404f3c1062dc)

Whats werid:
What prevented me from downgrading to sway 1.2 was this error when starting sway:
```
sway: error while loading shared libaries: lib.json-c.so.4: cannot open shared object file: No such file or directory
```
I had json-c version-0.14 (which was not working) and downgraded to json-c version 0.13.1, which provided the missing file.
[Link to json-c releases](https://github.com/json-c/json-c/releases)

the latest sway 1.4 release (i have dowloaded and doesn't work) need json-c version 0.14 (and doesnt work with json-c version 0.13.1)
the current master also need json-c version 0.13.1, because with version 0.14 sway throws:
```
sway: error while loading shared libaries: lib.json-c.so.5: cannot open shared object file: No such file or directory
```

TLDR:
sway master is working, but needs json-c version 0.14.
lastest sway 1.4 release needs json-c version 0.13.1.
sway 1.2 also needs version 0.14
So why does the current master needs the older json-c ?


Unfortunatly downgrading to json-c 0.13.1 breaks other programs

**Comment by FreeFull** [comment]
Current master needs whatever json-c you had installed when you compiled it. Sounds like you had 0.13.1 installed, then compiled sway-git, then tried installing 0.14. If you recompile sway-git with json-c 0.14 installed, it should work.

**Comment by theErikss0n** [comment]
Thank your @FreeFull, this worked.

To conclude:
All 3 monitors work with the current master and `WLR_DRM_NO_MODIFIERS=1`.
Thanks for your help

**Comment by emersion** [comment]
Okay, nice. So the only thing that remains to be done is to make it work without `WLR_DRM_NO_MODIFIERS` set. For this reason I'm going to close this issue as a duplicate of https://github.com/swaywm/wlroots/issues/1877.
